cpp/frameProcessor/src/LiveViewPlugin.cpp

```diff
@@ -113,7 +123,14 @@ void LiveViewPlugin::process_frame(boost::shared_ptr<Frame> frame)
         }
 
         //Count all frames that match the dataset(s) and tag(s)
-        frame_count_ ++;
+        if(dataset == datasets_.end())
+        {
+          frame_count_ ++;
+        }
+        else
+        {
+          dataset->second ++;
+        }
```

Comment:
This could be removed if you do the increment in the if-else above, e.g.

```suggestion
        count = dataset->second++;
```

which will assign the current value to count and then increment.